_edgedb_go/main.go

```diff
@@ -119,6 +122,90 @@ func http_worker(
 	}
 }
 
+func edgedb_worker(
+	start time.Time, duration, timeout time.Duration,
+	query string, idStrings []string, wg *sync.WaitGroup,
+	report ReportFunc,
+) {
+	defer wg.Done()
+
+	var (
+		err         error
+		min_latency int64 = math.MaxInt64
+		max_latency int64 = 0
+		queries     int64 = 0
+	)
+
+	samples := make([]string, 0, *nsamples)
+	latency_stats := make([]int64, timeout/1000/10+1)
+	timeout_ns := timeout.Nanoseconds()
+
+	num_ids := len(idStrings)
+	ids := make([]types.UUID, num_ids)
+	for i := 0; i < len(idStrings); i++ {
+		ids[i], err = types.UUIDFromString(idStrings[i])
+		if err != nil {
+			log.Fatal(err)
+		}
+	}
+
+	ctx := context.TODO()
+	client, err := edgedb.Connect(ctx, edgedb.Options{
+		Host:     *host,
+		Port:     *port,
+		User:     "edgedb",
+		Database: "edgedb_bench",
+	})
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	defer client.Close()
+
+	args := make(map[string]interface{}, 1)
+
+	for time.Since(start) < duration || duration == 0 {
+		args["id"] = ids[rand.Intn(num_ids)]
+
+		req_start := time.Now()
+		resp, err := client.QueryOneJSON(ctx, query, args)
```

Comment:
So for now this is only benchmarking JSON. We'll also need to implement https://github.com/edgedb/webapp-bench/blob/master/_edgedb/queries_repack.py in Go -- that effectively tests how fast the codecs pipeline is.

Reply:
Should that be part of the same PR?

Reply:
Not necessarily, we can merge this. I'm only not sure about the `git config --global url.git@github.com:.insteadOf https://github.com/` part. It would be great if `make go` would do some magic without requiring messing with `git config --global`. Is there some way?